src/main/java/com/github/checkstyle/regression/module/ModuleUtils.java

```diff
@@ -0,0 +1,172 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import com.github.checkstyle.regression.data.GitChange;
+import com.github.checkstyle.regression.data.ModuleExtractInfo;
+
+/**
+ * Contains utility methods related to checkstyle module.
+ * @author LuoLiangchen
+ */
+public final class ModuleUtils {
+    /** The compiled regex pattern of the path of Java source files. */
+    private static final Pattern JAVA_SOURCE_PARTTEN =
+            Pattern.compile("src/(main|test)/java/(.+)\\.java");
+
+    /** The postfix of a test of a checkstyle module. */
+    private static final String TEST_POSTFIX = "Test";
+
+    /** The map of full qualified name to module extract info. */
+    private static final Map<String, ModuleExtractInfo> NAME_TO_MODULE_EXTRAT_INFO =
+            new HashMap<>();
+
+    /** Prevents instantiation. */
+    private ModuleUtils() {
+    }
+
+    /**
+     * Sets the map of full qualified name to module extract info with the given map.
+     * @param map the given map
+     */
+    public static void setNameToModuleExtratInfo(Map<String, ModuleExtractInfo> map) {
+        NAME_TO_MODULE_EXTRAT_INFO.clear();
+        NAME_TO_MODULE_EXTRAT_INFO.putAll(map);
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a checkstyle module.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a checkstyle module
+     */
+    public static boolean isCheckstyleModule(GitChange change) {
+        final boolean returnValue;
+        if (isJavaMainSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change.getPath());
+            returnValue = NAME_TO_MODULE_EXTRAT_INFO.containsKey(fullName);
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a checkstyle utility class.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a checkstyle utility class.
+     */
+    public static boolean isCheckstyleUtility(GitChange change) {
+        final boolean returnValue;
+        if (isJavaMainSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change.getPath());
+            returnValue = !NAME_TO_MODULE_EXTRAT_INFO.containsKey(fullName);
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a Java main source file.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
```

Comment:
I add some info in each main description. Refers to the similar utility methods in `ModuleReflectionUtils` in checkstyle project.